chain-impl-mockchain/src/vote/manager.rs

```diff
@@ -158,6 +171,88 @@ impl ProposalManager {
         })
     }
 
+    #[must_use = "Compute the PrivateTally in a new ProposalManager, does not modify self"]
+    pub fn private_tally(&self, stake: &StakeControl) -> Result<Self, VoteError> {
```

Comment:
Would it make sense to move `self` into this and other similar methods below?
Is the extant `ProposalValue` instance of any use, so that all the cloning is necessary?